test/catch2_test_block_shuffle.cu

```diff
@@ -175,45 +183,62 @@ struct down_with_prefix_op_t
 
     block_shuffle.Down(thread_data, thread_data, prefix);
 
-    if (static_cast<int>(threadIdx.x) == m_target_thread_id)
+    if (cub::RowMajorTid(blockDim.x, blockDim.y, blockDim.z) == m_target_thread_id)
     {
       m_d_prefix_ptr[0] = prefix;
     }
   }
 };
 
 template <int ItemsPerThread,
-          int ThreadsInBlock,
+          int BlockDimX,
+          int BlockDimY,
+          int BlockDimZ,
           class T,
           class ActionT>
 void block_shuffle(
     thrust::device_vector<T> &data, 
     ActionT action)
 {
-  block_shuffle_kernel<ThreadsInBlock, ItemsPerThread>
-    <<<1, ThreadsInBlock>>>(
+  dim3 block(BlockDimX, BlockDimY, BlockDimZ);
+  block_shuffle_kernel<BlockDimX, BlockDimY, BlockDimZ, ItemsPerThread>
+    <<<1, block>>>(
         thrust::raw_pointer_cast(data.data()),
         action);
 
   REQUIRE( cudaSuccess == cudaPeekAtLastError() );
   REQUIRE( cudaSuccess == cudaDeviceSynchronize() );
 }
 
-// %PARAM% THREADS_IN_BLOCK bs 64:512:1024
+// %PARAM% MULTI_DIM mdim 0:1
+// %PARAM% CONFIG cfig 0:1:2
+
+using multi_dim_x   = c2h::enum_type_list<int, 7, 32, 64>;
+using multi_dim_yz  = c2h::enum_type_list<int, 2, 2, 2>;
+
+using single_dim_x  = c2h::enum_type_list<int, 64, 512, 1024>;
+using single_dim_yz = c2h::enum_type_list<int, 1, 1, 1>;
```

Comment:
This is solution is quite expensive. Could you make it `c2h::enum_type_list<int, 1>` instead? Conditionals are not great as well. Is there anything stopping you from doing something like:
```cpp
#if MULTI_DIM
using block_dim_xs = c2h::enum_type_list<int, 7, 32, 64>;
#else 
using block_dim_xs = c2h::enum_type_list<int, 64, 512, 1024>;
#endif
```